modules/flye/main.nf

```diff
@@ -0,0 +1,58 @@
+process FLYE {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::flye=2.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/flye:2.9--py39h6935b12_1' :
+        'quay.io/biocontainers/flye:2.9--py39h6935b12_1' }"
+
+    input:
+    tuple val(meta), path(reads)
+    val mode
+
+    output:
+    tuple val(meta), path("*.fasta.gz"), emit: fasta
+    tuple val(meta), path("*.gfa.gz")  , emit: gfa
+    tuple val(meta), path("*.gv.gz")   , emit: gv
+    tuple val(meta), path("*.txt")     , emit: txt
+    tuple val(meta), path("*.log")     , emit: log
+    tuple val(meta), path("*.json")    , emit: json
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    flye $mode $reads --out-dir . --threads $task.cpus $args
+
+    gzip -c assembly.fasta > ${prefix}.assembly.fasta.gz
+    gzip -c assembly_graph.gfa > ${prefix}.assembly_graph.gfa.gz
+    gzip -c assembly_graph.gv > ${prefix}.assembly_graph.gv.gz
+    mv assembly_info.txt ${prefix}.assembly_info.txt
+    mv flye.log ${prefix}.flye.log
+    mv params.json ${prefix}.params.json
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        flye: \$(echo \$(flye --version | sed 's/-b1768//' ))
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo stub > assembly.fasta | gzip -c assembly.fasta > ${prefix}.assembly.fasta.gz
+    echo stub > assembly_graph.gfa | gzip -c assembly_graph.gfa > ${prefix}.assembly_graph.gfa.gz
+    echo stub > assembly_graph.gv | gzip -c assembly_graph.gv > ${prefix}.assembly_graph.gv.gz
+    echo contig_1 > ${prefix}.assembly_info.txt
+    echo stub > ${prefix}.flye.log
+    echo stub > ${prefix}.params.json
+
+    echo 'FLYE:' > versions.yml
+    echo ' flye: 2.9' >> versions.yml
```

Comment:
Is it possible to circumvent hardcoding the version?

Reply:
Sure! I thought I would do a real stub run so flye shouldn't be executed at all

Reply:
I would say here to keep using the same syntax as above (the HEREDOC). The stub should only circumvent the long running stuff.